boards/wio_terminal/src/wifi/wifi_rpcs.rs

```diff
@@ -0,0 +1,396 @@
+                    .field("type", &self.bss_type)
+                    .field("security", &self.security)
+                    .field("wps", &self.wps)
+                    .field("channel", &self.chan)
+                    .field("band", &self.band)
+                    .finish()
+            } else {
+                f.debug_struct("ScanResult")
+                    .field("bssid", &self.bssid)
+                    .field("rssi", &self.rssi)
+                    .field("type", &self.bss_type)
+                    .field("security", &self.security)
+                    .field("wps", &self.wps)
+                    .field("channel", &self.chan)
+                    .field("band", &self.band)
+                    .finish()
+            }
+        }
+    }
+}
+
+impl Default for ScanResult {
+    fn default() -> Self {
+        Self {
+            ssid: SSID {
+                len: 0,
+                value: [0u8; 33],
+            },
+            bssid: BSSID([0u8; 6]),
+            rssi: 0,
+            bss_type: BssType::Any,
+            security: Security::Open,
+            wps: WPS::Default,
+            chan: 0,
+            band: Band::_24Ghz,
+        }
+    }
+}
+
+pub struct ScanGetAP<N: ArrayLength<ScanResult>> {
+    m: core::marker::PhantomData<N>,
+}
+
+impl<N: ArrayLength<ScanResult>> ScanGetAP<N> {
+    pub fn new() -> Self {
+        Self {
+            m: core::marker::PhantomData,
+        }
+    }
+}
+
+impl<N: ArrayLength<ScanResult>> codec::RPC for ScanGetAP<N> {
+    type ReturnValue = (GenericArray<ScanResult, N>, i32);
+    type Error = usize;
+
+    fn header(&self, seq: u32) -> codec::Header {
+        codec::Header {
+            sequence: seq,
+            msg_type: id::MsgType::Invocation,
+            service: id::Service::Wifi,
+            request: id::WifiRequest::ScanGetAP.into(),
+        }
+    }
+
+    fn args(&self, buff: &mut heapless::Vec<u8, heapless::consts::U64>) {
+        let num = N::to_u16().to_le_bytes();
+        buff.extend_from_slice(&num).ok();
+    }
+
+    fn parse(&mut self, data: &[u8]) -> Result<Self::ReturnValue, Err<Self::Error>> {
+        let (data, hdr) = codec::Header::parse(data)?;
+        if hdr.msg_type != id::MsgType::Reply
+            || hdr.service != id::Service::Wifi
+            || hdr.request != id::WifiRequest::ScanGetAP.into()
+        {
+            return Err(Err::NotOurs);
+        }
+
+        let (data, l) = streaming::le_u32(data)?; // Binary len - returning 62 bytes per result
+        if l as usize != (core::mem::size_of::<ScanResult>() * N::to_usize()) {
+            return Err(Err::ResponseOverrun);
+        }
+
+        let mut res = GenericArray::<ScanResult, N>::default();
+
+        //let (data, ssid) = SSID::parse(data)?;
+        for i in 0..N::to_usize() {
+            res[i] = unsafe { *((data.as_ptr() as *const ScanResult).offset(i as isize)) };
+        }
+
+        //let s: ScanResult = unsafe { core::mem::transmute::<[u8;
```

Comment:
There's some commented out code here. Did you forget or is it still a work in progress?

Reply:
Fixed (but in different crate now).